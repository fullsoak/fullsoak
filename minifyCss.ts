import { default as CleanCss } from "clean-css";
const cssCleaner = new CleanCss({});

export const cleanCss = (css?: string): string => cssCleaner.minify(css).styles;
