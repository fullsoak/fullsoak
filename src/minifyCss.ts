import { default as CleanCss } from "clean-css";
import { LogWarn } from "./utils.ts";
const cssCleaner = new CleanCss({
  inline: ["local", "remote"], // @TODO make `remote` configurable
});

/**
 * utility function, takes in the raw css string and outputs a minified version of it
 */
export const cleanCss = (css?: string): string => {
  let retVal = "";
  const output = cssCleaner.minify(css) || {};
  retVal = output.styles || "";
  if (output.errors?.length) {
    LogWarn("error while minifying css", JSON.stringify(output.errors));
  }
  // @TODO log both errors and warnings, potentially depending on
  // user-specified log level
  return retVal;
};
