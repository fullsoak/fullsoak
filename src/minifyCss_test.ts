import { assertSnapshot } from "@std/testing/snapshot";
import { cleanCss } from "./minifyCss.ts";

Deno.test("minifyCss", async (t) => {
  const css = `html, body {
    padding: 0;
    margin: 0;
  }`;
  const cleanedCss = cleanCss(css);
  await assertSnapshot(t, cleanedCss);
});
