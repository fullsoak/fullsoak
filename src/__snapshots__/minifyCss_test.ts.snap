export const snapshot = {};

snapshot[`minifyCss 1`] = `"body,html{padding:0;margin:0}"`;
